lib/spack/spack/compiler.py

```diff
@@ -297,11 +297,17 @@ def verify_executables(self):
         Raises a CompilerAccessError if any of the non-null paths for the
         compiler are not accessible.
         """
-        missing = [cmp for cmp in (self.cc, self.cxx, self.f77, self.fc)
-                   if cmp and not (os.path.isfile(cmp) and
-                                   os.access(cmp, os.X_OK))]
-        if missing:
-            raise CompilerAccessError(self, missing)
+        def accessible_exe(exe):
+            # Cray compilers may not have absolute paths
```

Comment:
Other compilers could do this as well, couldn't they? I think this could be rephrased like

> convert relative paths to absolute paths

although I don't think that's actually true: this probably only handles an exe name or an absolute path so something like

> convert exe names to full paths

would be best.

Also, the `which_string` method already does the `isfile` and `os.access` checks.

Reply:
I realized `which_string` is not as close to the `which` utility as I had hoped (in terms of how it handles absolute paths). That redundancy will go away with my next push.

Reply:
I don't think non-Cray compilers will ever do this naturally, but I suppose a user could hand-edit it and my comment will account for that.